strings/base_collections_map.h

```diff
@@ -2,13 +2,17 @@
 namespace winrt::impl
 {
     template <typename K, typename V, typename Container>
-    struct observable_map :
```

Comment:
Should the lock type or lock instance be configurable?

Reply:
I'd like to avoid feature creep unless there's a compelling need for this at the moment. SRW locks are fantastic and making it configurable won't necessarily make it usable by those with "interesting" locking requirements.